java-frontend/src/main/java/org/sonar/java/model/pattern/GuardedPatternTreeImpl.java

```diff
@@ -57,7 +57,7 @@ public PatternTree pattern() {
   }
 
   @Override
-  public SyntaxToken andOperator() {
+  public SyntaxToken whenOperator() {
```

Comment:
The corresponding field should be renamed as well (as well as the constructor parameter)